src/bootstrap/compile.rs

```diff
@@ -961,8 +961,12 @@ pub fn stream_cargo(
     // stderr as piped so we can get those pretty colors.
     let mut message_format = String::from("json-render-diagnostics");
     if let Some(s) = &builder.config.rustc_error_format  {
-        message_format.push_str(",json-diagnostic-");
-        message_format.push_str(s);
+        if s == "json" {
+            message_format = String::from("json,json-diagnostic-rendered-ansi");
+        } else {
+            message_format.push_str(",json-diagnostic-");
+            message_format.push_str(s);
+        }
```

Comment:
What went wrong with this diff?

Reply:
oops, sorry, this was not meant to be in this PR